Header.hpp

```diff
@@ -3,30 +3,294 @@
+    QString base_model;
+    int device_id =0;
+    bool draw_bounding_boxes = false;
+    bool draw_skeleton_edges = false;
+    QVector<Size> input_resolution = {{0, 0}};
+    float min_keypoint_confidence = 0.0f;
+    float min_pose_confidence = 0.0f;
+    bool multi_pose_detection = false;
+    int output_stride = 0;
 
-    std::strong_ordering operator<=>(const MyObject &) const = default; // Provides a strong ordering comparison operator (<=>) for comparing two MyObject instances
+    bool operator==(const Posenet_params& other) const noexcept = default;
+};
+
+struct Trt_params {
+    Q_GADGET
+    Q_PROPERTY(bool optimize MEMBER optimize)
+    Q_PROPERTY(QString keypoint_dataset MEMBER keypoint_dataset)
+    Q_PROPERTY(QString model_path MEMBER model_path)
+    Q_PROPERTY(QString model_optimized MEMBER model_optimized)
+    Q_PROPERTY(QString model_weights MEMBER model_weights)
+    Q_PROPERTY(QString model_desc MEMBER model_desc)
+    Q_PROPERTY(int model_height MEMBER model_height)
+    Q_PROPERTY(int model_width MEMBER model_width)
+    Q_PROPERTY(float min_pose_confidence MEMBER min_pose_confidence)
+
+public:
+    bool optimize = false;
+    QString keypoint_dataset;
+    QString model_path;
+    QString model_optimized;
+    QString model_weights;
+    QString model_desc;
+    int model_height = 0;
+    int model_width = 0;
+    float min_pose_confidence = 0.0f;
+
+    bool operator==(const Trt_params& other) const noexcept = default;
+};
+
+
+//pose_backends array of type variant
+struct Pose_backend {
+    Q_GADGET
+    Q_PROPERTY(QString name MEMBER name)
+    Q_PROPERTY(QVariant params MEMBER params)
+
+public:
+    QString name;
+    QVariant params; //either of the three types
+
+    bool operator==(const Pose_backend& other) const noexcept = default;
+};
+
+//camera parameters
+struct Camera_params {
+    Q_GADGET
+    Q_PROPERTY(Size resolution MEMBER resolution)
+    Q_PROPERTY(int framerate MEMBER framerate)
+    Q_PROPERTY(bool flip MEMBER flip)
+    Q_PROPERTY(int rotate MEMBER rotate)
+
+public:
+    Size resolution = {0, 0};
+    int framerate = 0;
+    bool flip = false;
+    int rotate = 0;
+
+    bool operator==(const Camera_params& other) const noexcept = default;
+};
+
+struct Cameras {
+    Q_GADGET
+    Q_PROPERTY(QVector<QString> input_paths MEMBER input_paths)
+    Q_PROPERTY(bool flip_camera MEMBER flip_camera)
+    Q_PROPERTY(QMap<QString, Camera_params> params MEMBER params)
+    Q_PROPERTY(QMap<QString, QString> intrinsics MEMBER intrinsics)
+
+public:
+    QVector<QString> input_paths;
+    bool flip_camera = false;
+    QMap<QString, Camera_params> params;
+    QMap<QString, QString> intrinsics; //idk what the structure is here
+
+    bool operator==(const Cameras& other) const noexcept = default;
+};
+
+//dimmaps 
+struct Pose2DToFloor {
+    Q_GADGET
+    Q_PROPERTY(QMap<QString, QVector<QVector<float>>> pairs_per_camera MEMBER pairs_per_camera)
```

Comment:
I would recommend typedef'ing such a type : 

    using cameras_pair_type = QMap<QString, QVector<QVector<float>>>;
    
also I wonder about QVector<QVector<float>> : what's the format of the data like ? 
If it's e.g. always a list of two / three elements it would be better to have a "point 2D" / "point 3D" type (there are already such types in Qt, e.g. QVector2D, QVector3D).
If every element can have a different length then what you did is the correct thing

Reply:
ok i don't really know the context / whether this will change depending on the camera so i assumed it will change

The only dimmap obj example i found is here: https://gitlab.com/sat-mtl/tools/livepose/livepose/-/blob/main/livepose/configs/mediapipe.json 

Should i ask Manuel?

Reply:
:+1: I saw you asked him, what's the conclusion?

Reply:
Added a new PointPair struct!  

For adding something like this: using cameras_pair_type = QMap<QString, QVector<QVector<float>>>;  

it's just for simplifying / making it easy to read, right?

Reply:
yes

Reply:
the more precise the naming, the better :)